grace/models/feature_extractor.py

```diff
@@ -135,20 +137,41 @@ def forward(
 
             if (
                 x_low
-                >= image_shape[-1]
+                > image_shape[-1]
                 - self.bbox_size[0] * self.keep_patch_fraction
                 or x_low + self.bbox_size[0]
                 < self.bbox_size[0] * self.keep_patch_fraction
                 or y_low
-                >= image_shape[-2]
+                > image_shape[-2]
                 - self.bbox_size[1] * self.keep_patch_fraction
                 or y_low + self.bbox_size[0]
                 < self.bbox_size[1] * self.keep_patch_fraction
             ):
                 node_attrs[GraphAttrs.NODE_GROUND_TRUTH] = Annotation.UNKNOWN
-                continue
```

Comment:
Works now, but we need to make sure that the corner padding is identical to the random edge cropping when augmenting... I raised an issue for future brainstorming